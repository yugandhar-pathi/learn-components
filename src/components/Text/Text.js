import PropTypes from 'prop-types'
import {TextStyle} from './Text.styles'


export const Text = ({ size, dataTestId, children }) => {
  
    return (
        <TextStyle data-testid={dataTestId} size={size}>{children}</TextStyle>
    )
}

Text.propTypes = {
    /*Text to be printed*/
    children: PropTypes.string.isRequired,
    size: PropTypes.oneOf(["small","medium","large"]),
    dataTestId: PropTypes.string.isRequired,
}

Text.defaultProps = {
    size: "medium"
}

