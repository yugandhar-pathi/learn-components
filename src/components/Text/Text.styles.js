import styled from 'styled-components'

const TextSize = {
    small: '18px',
    medium: '22px',
    large: '26px'
}


export const TextStyle = styled.p`
   font-size: ${(props) => TextSize[props.size]};
   line-height: ${(props) => TextSize[props.size]};
`