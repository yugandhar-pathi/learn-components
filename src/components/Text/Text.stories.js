import React from 'react'
import { Text } from './Text';

export default {
    component: Text,
};

const Template = (args) => <Text {...args} />;


export const Default = Template.bind({});
Default.args = {
    children: "I am Medium size text",
};

export const Small = Template.bind({});
Small.args = {
    children: "I am Small size text",
    size: 'small',
};

export const Large = Template.bind({});
Large.args = {
    children: "I am Large size text",
    size: 'large',
};