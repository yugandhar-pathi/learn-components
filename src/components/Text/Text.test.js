import {render, screen} from '@testing-library/react'
import {Text} from './Text'


test("Large Button",() => {
    render(<Text size="large" dataTestId='large-text'>Large</Text>)

    expect(screen.getByTestId('large-text').textContent).toBe('Large')
})

test("Snapshot Small Button",() => {
    render(<Text size="large" dataTestId='large-text'>Small</Text>)

    expect(screen.getByTestId('large-text')).toMatchSnapshot()
})

test("Snapshot medium Button",() => {
    render(<Text dataTestId='large-text'>medium</Text>)

    expect(screen.getByTestId('large-text')).toMatchSnapshot()
})