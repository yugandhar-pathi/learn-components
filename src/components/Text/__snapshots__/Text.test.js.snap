// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Small Button 1`] = `
<p
  class="sc-bczRLJ dWpahr"
  data-testid="large-text"
>
  Small
</p>
`;

exports[`Snapshot medium Button 1`] = `
<p
  class="sc-bczRLJ jjGmFX"
  data-testid="large-text"
>
  medium
</p>
`;
